.travis.yml

```diff
@@ -3,8 +3,7 @@ language: php
 php:
   - 5.6
   - 7.0
-  - 7.1
-  - 7.2
```

Comment:
I would prefer to keep 7.2 over 7.0 which is soon to be eol.

Reply:
why? we need the upper and lower limits of the language per major.
rest in between should be fine based on this.

Reply:
Ok. The way typehints works between 7.0, 7.1 and 7.2 is different enough that we should keep 7.1 and 7.2 as there are changes that could be made where code is valid in 7.2 but broken in 7.1.

Reply:
But it would be the same for 7.3 if it changes after another minor. The changes are always linear here, not fluctuating.
Can you show be one example that would not be true for the case as I outline it?

Reply:
PHP 7.2+ allows parameter widening, while 7.1 does not. https://3v4l.org/oISbn

```php
class Fruit {
    public function eat(Mammal $h) {
    }
}

class Apple extends Fruit {
    public function eat($h) {   
    }
}

class Mammal {
}
```

Reply:
Yeah, but why wouldnt that be covered automatically by 7.3 then? Thats my point.

Reply:
The min+max always works here for our use cases: https://github.com/dereuromark/cakephp-ide-helper/blob/master/.travis.yml#L6-L7
It can be sprinkled with coverage as `7.0` or alike to have also a min for the major - but thats all optional.

Reply:
> PHP 7.2+ allows parameter widening, while 7.1 does not. https://3v4l.org/oISbn
> 
> ```
> class Fruit {
>     public function eat(Mammal $h) {
>     }
> }
> 
> class Apple extends Fruit {
>     public function eat($h) {   
>     }
> }
> 
> class Mammal {
> }
> ```

But so does 7.1 - doesn't it?

Reply:
What will CakePHP target anyway? 7.2, 7.3?

Reply:
all declared in composer json file.

Reply:
> Yeah, but why wouldnt that be covered automatically by 7.3 then? Thats my point. 

Yes 7.2 and 7.3 behave the same here.  I guess 7.0 and 7.1 also behave the same. It seems strange to me to pick 7.0 over 7.1 though. But I guess it won't matter much until it does :smile: